core/src/com/unciv/ui/worldscreen/BackgroundActor.kt

```diff
@@ -0,0 +1,61 @@
+package com.unciv.ui.worldscreen
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.graphics.g2d.Batch
+import com.badlogic.gdx.graphics.g2d.NinePatch
+import com.badlogic.gdx.scenes.scene2d.Actor
+import com.badlogic.gdx.scenes.scene2d.Touchable
+import com.badlogic.gdx.scenes.scene2d.utils.Drawable
+import com.badlogic.gdx.scenes.scene2d.utils.NinePatchDrawable
+import com.badlogic.gdx.utils.Align
+import com.unciv.ui.images.ImageGetter
+
+/** An Actor that just draws a Drawable [background], preferably a [NinePatchDrawable] created
+ *  by [BackgroundActor.getRoundedEdgeRectangle], meant to work in Table Cells and to be overlaid with other Widgets.
+ *  The drawable's center can be moved to any of the corners or vertex centers using `align`, which will also scale the
+ *  drawable up by factor 2 and clip to the original rectangle. This can be used to draw rectangles with one or two rounded corners.
+ *  @param align An [Align] constant - In which corner of the [BackgroundActor] rectangle the center of the [background] should be.
+ */
+class BackgroundActor(val background: Drawable, align: Int) : Actor() {
+    private val widthMultiplier = if (Align.isCenterHorizontal(align)) 1f else 2f
+    private val heightMultiplier = if (Align.isCenterVertical(align)) 1f else 2f
+    private val noClip = (align and Align.center.inv()) == 0
```

Comment:
This is obscure, don't use boolean logic on ints, even if it does work

Reply:
The only test I wanted that Java hasn't in Align... :sob:  - 

but OK if we do not want to be hardened against align values Align.center and 0 being in fact equivalent to all other code we can just compare directly, much simpler. _Or_ Align.isCenterHorizontal(align) && Align.isCenterVertical(align) would be the same thing...

The whole class is not used for all it _can_ do - which this test showed: 
<details><summary>Screen</summary>

![image](https://user-images.githubusercontent.com/63000004/170885745-4da3f523-5ee5-4ee9-963c-e016993e8c5c.png)
</details>
... but maybe we want such building blocks elsewhere in the future. That test helped a lot to think about the logic correctly.